src/composables/map/street-view.composable.ts

```diff
@@ -0,0 +1,257 @@
+import { watch, nextTick } from 'vue'
+import { storeToRefs } from 'pinia'
+import useMap from '@/composables/map/map.composable'
+import { useAppStore } from '@/stores/app.store'
+import { useInfoStore } from '@/stores/info.store'
+import { MapBrowserEvent } from 'ol'
+import BaseEvent from 'ol/events/Event'
+import { Coordinate } from 'ol/coordinate'
+import { Select } from 'ol/interaction'
+import { fromLonLat, toLonLat } from 'ol/proj'
+import { containsCoordinate } from 'ol/extent.js'
+import { loadGoogleapis } from '@/services/info/street-view'
+import { SvCompassFeature } from '@/services/info/sv-compass-feature'
+import { SvDirectionFeature } from '@/services/info/sv-direction-feature'
+
+import { Feature } from 'ol'
+import { Point } from 'ol/geom'
+import VectorLayer from 'ol/layer/Vector'
+import VectorSource from 'ol/source/Vector'
+
+export const DEFAULT_INFO_ZINDEX = 1500
+export const SV_FEATURE_LAYER_TYPE = 'svFeatureLayer'
+
+const SV_RADIUS = 90
+
+export default function useStreeView() {
+  const { infoOpen } = storeToRefs(useAppStore())
+  const {
+    locationInfo,
+    hidePointer,
+    isStreetviewActive,
+    noDataAtLocation,
+    panoPositionChanging,
+    svFeature,
+  } = storeToRefs(useInfoStore())
+
+  const map = useMap().getOlMap()
+
+  let googleService: any = null
+  let streetViewService: any = null
+  let panorama: any = null
+  let panoramaLinksListener: any = null
+  let panoramaPovListener: any = null
+  const selectFeature = new Select({
+    filter: (feature /*, layer*/) => feature instanceof SvDirectionFeature,
+  })
+  selectFeature.on('select', handleNavigate)
+
+  const svPoint = new Point([0, 0])
+  const svCompassFeature = new SvCompassFeature()
+  svCompassFeature.setGeometry(svPoint)
+
+  const svFeatureLayer = new VectorLayer({
+    source: new VectorSource({
+      features: [] as Feature[],
+    }),
+    // altitudeMode: 'clampToGround',
+    zIndex: DEFAULT_INFO_ZINDEX,
+  })
+  svFeatureLayer.set('cyLayerType', SV_FEATURE_LAYER_TYPE)
+  map.addLayer(svFeatureLayer)
+
+  function setLocation(loc: Coordinate | undefined) {
+    if (panorama !== null) {
+      if (loc && !panoPositionChanging.value) {
+        const lonlat = toLonLat(loc, map.getView().getProjection())
+        streetViewService.getPanorama(
+          {
+            location: {
+              lat: lonlat[1],
+              lng: lonlat[0],
+            },
+            radius: SV_RADIUS,
+          },
+          updatePanorama
+        )
+      }
+    }
+  }
+
+  function updatePanorama(data: any, status: any) {
```

Comment:
any?

Reply:
no types available for google services loaded on-demand

Reply:
soooooooo????? you can't define one?

Reply:
It's mostly not knowing how to do it.
I found a lib for type definitions (npm i -D @types/google.maps)
I'll try if this helps.